python/pyspark/sql/functions.py

```diff
@@ -2044,7 +2044,7 @@ class UserDefinedFunction(object):
 
     .. versionadded:: 1.3
     """
-    def __init__(self, func, returnType, name=None, vectorized=False):
+    def __init__(self, func, returnType, name=None, vectorized=False, grouped=False):
```

Comment:
`vectorized=False, grouped=True` is an invalid combination. How about we introduce a `udfType` and `0` means normal udf, `1` means pandas udf, and `2` means pandas grouped udf?  We can create something like `object PythonEvalType` to sync this encoding between python and java.

Reply:
Sounds good. I'll modify it.